compiler/rustc_resolve/src/imports.rs

```diff
@@ -297,6 +297,29 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         })
     }
 
+    /// find the nearest ambiguity and update it to warning
```

Comment:
Ah, that's why you originally placed the flag into `NameBinding`, it applied not just to the current `NameBinding` but to the whole import chain starting from it.
It was a better decision, let's revert to that.